extensions/amp-experiment/1.0/attribute-allow-list/attribute-allow-list.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  DefaultAllowedURLAttributeMutationEntry,
+  DefaultStyleAllowedAttributeMutationEntry,
+} from './allowed-attribute-mutation-entry';
+import {user, userAssert} from '../../../../src/log';
+
+const TAG = 'amp-experiment allowed-mutations';
+
+/**
+ * Function to validate the attribute mutation
+ * should be allowed, and return its mutation
+ * @param {!Object} mutationRecord
+ * @param {string} stringifiedMutation
+ * @return {!./allowed-attribute-mutation-entry.AllowedAttributeMutationEntry}
+ */
+export function getAllowedAttributeMutationEntry(
+  mutationRecord,
+  stringifiedMutation
+) {
+  // Assert the mutation attribute is one of the following keys
+  const mutationAttributeName = mutationRecord['attributeName'];
+  userAssert(
+    attributeMutationAllowList[mutationAttributeName] !== undefined,
+    'Mutation %s has an unsupported attributeName.',
+    stringifiedMutation
+  );
+
+  // Find our allow list entry
+  const mutationTagName = mutationRecord['targetElement'].tagName.toLowerCase();
+
+  // Search through the allow list for our
+  // Allowed attribute entry
+  let allowedAttributeMutationEntry = undefined;
+  if (attributeMutationAllowList[mutationAttributeName][mutationTagName]) {
```

Comment:
I'm a bit concerned about the number of entries of the `attributeMutationAllowList[mutationAttributeName]` if there're many `mutationTagNames` that we support. For example `src` maybe allowed for a list of tags, are we going to have an instance for each of them.

Reply:
Yes, though the list of `src` supported elements is small: https://www.w3schools.com/tags/att_src.asp

Src is an interesting case though, as for not, it should only be amp-img.

In my opinion, even in the worst case scenario of everything being done on the client, I don't think this list would get extremely large if we do it an on a request basis. Especially since we are implementing '*' values for tag names.

Let me know if this makes sense.  I feel that is more of a design questions. In my opinion, Allow list offer security and verbosity, at the expense of being so verbose. We can take this offline if you think it helps 👍